client_app/src/components/volunteer/ConfirmationPage.js

```diff
@@ -3,12 +3,28 @@ import { Link } from "react-router-dom";
 import IconButton from "@mui/material/IconButton";
 import Tooltip from "@mui/material/Tooltip";
 import InfoIcon from "@mui/icons-material/Info";
+import CancelIcon from "@mui/icons-material/Cancel";
 import CheckCircleIcon from "@mui/icons-material/CheckCircle";
 
 const handleReload = () => {
   window.location.reload();
 };
 
+
+const checkForOverlap = (shift, otherShifts) => {
+  return otherShifts.some(otherShift => {
```

Comment:
you should be able to just directly compare the shift.start_time and shift.end_time